===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ PortObjectAppendPortObject:n$11 == { -1 } }) free(*(PortObjectAppendPortObject:prid2)) at 6 (line 1491, column 9) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/snort-2.9.13/src/sfutil/sfportobject.c
+++ b/table1/benchmarks/snort-2.9.13/src/sfutil/sfportobject.c
@@ -1494,7 +1494,8 @@ PortObject * PortObjectAppendPortObject(PortObject * poa, PortObject * pob )
        if( !prid2 )
            return 0;
        *prid2 = *prid;
-       sflist_add_tail(poa->rule_list,prid2);
+       if (sflist_add_tail(poa->rule_list,prid2) == -1)
+          free(prid2);
    }
    return poa;
 }
